src/theory/sets/solver_state.cpp

```diff
@@ -98,8 +99,6 @@ void SolverState::registerTerm(Node r, TypeNode tnn, Node n)
   {
     if (nk == SINGLETON)
     {
-      // singleton lemma
-      getProxy(n);
```

Comment:
This call here to getProxy was generating a proxy for kinds: EMPTYSET, INTERSECTION, SETMINUS, UNION, UNIVERSE_SET, and SINGLETON. 
Now proxy are generated only for SINGLETON in TheorySetsPrivate::fullEffortCheck() line [364](https://github.com/CVC4/CVC4/pull/5037/files#diff-aaa774b6c62e6fbcec9f9c49f7d60a3eR346). Does this mean the other kinds do not need proxy variables?

Reply:
This is only applied for SINGLETON (see guard directly above on line 99).  The behavior should be preserved, I believe.

Reply:
Lemmas are only generated for SINGLETON. But proxy variables are created for other kinds. 
See line https://github.com/CVC4/CVC4/blob/master/src/theory/sets/solver_state.cpp#L385

Reply:
This PR preserves the previous behavior for only calling `getProxy` for SINGLETON kinds during registerTerm (line 343 of the modified theory_sets_private.cpp).